src/mute/__tests__/mutedUsersReducer-test.js

```diff
@@ -0,0 +1,84 @@
+import Immutable from 'immutable';
+import deepFreeze from 'deep-freeze';
+
+import mutedUsersReducer from '../mutedUsersReducer';
+import { ACCOUNT_SWITCH, EVENT_MUTED_USERS, REALM_INIT } from '../../actionConstants';
+import { makeUserId } from '../../api/idTypes';
+
+describe('mutedUsersReducer', () => {
+  describe('REALM_INIT', () => {
+    test('when `muted_users` data is provided init state with it', () => {
+      const initialState = Immutable.Map({});
+      const action = deepFreeze({
+        type: REALM_INIT,
+        data: {
+          muted_users: [
+            {
+              id: 42,
+              timestamp: 1618822632,
+            },
+          ],
+        },
+      });
+
+      const actualState = mutedUsersReducer(initialState, action);
+
+      expect(actualState).toEqual(Immutable.Map([[makeUserId(42), 1618822632]]));
+    });
+
+    test('when no `muted_users` data is given reset state', () => {
+      const initialState = Immutable.Map([[makeUserId(42), 1618822632]]);
+      const action = deepFreeze({
+        type: REALM_INIT,
+        data: {},
+      });
+
+      const actualState = mutedUsersReducer(initialState, action);
+
+      expect(actualState).toEqual(Immutable.Map({}));
+    });
+  });
+
+  describe('ACCOUNT_SWITCH', () => {
+    test('resets state to initial state', () => {
+      const initialState = Immutable.Map([[makeUserId(42), 1618822632]]);
+
+      const action = deepFreeze({
+        type: ACCOUNT_SWITCH,
+      });
+
+      const actualState = mutedUsersReducer(initialState, action);
+
+      expect(actualState).toEqual(Immutable.Map({}));
+    });
+  });
+
+  describe('EVENT_MUTED_USERS', () => {
+    test('update `muted_users` when event comes in', () => {
+      const initialState = Immutable.Map([[makeUserId(42), 1618822632]]);
+
+      const action = deepFreeze({
+        type: EVENT_MUTED_USERS,
+        muted_users: [
+          {
+            id: 42,
+            timestamp: 1618822632,
+          },
+          {
+            id: 1234,
+            timestamp: 1618822635,
+          },
+        ],
+      });
+
+      const expectedState = Immutable.Map([
+        [makeUserId(42), 1618822632],
+        [makeUserId(1234), 1618822635],
+      ]);
+
+      const newState = mutedUsersReducer(initialState, action);
+
+      expect(newState).toEqual(expectedState);
+    });
+  });
```

Comment:
So this is a style that you'll see in a lot of our legacy tests, but I think the verbosity of it isn't helpful. It's possible to write all the same content with a lot less boilerplate; when we do, I think that (a) makes it easier to see what the test is really saying, and (b) makes it easier to scan through all the tests in the file and see what conditions are covered and think about what might not be.

For good examples to follow, you can look at tests written relatively recently by me or Chris. Scanning through `git log --stat -p --author 'greg\|cbobbe' -- src/**/*-test.js` and looking for newly-written tests (i.e. scanning past where we're just touching an existing test), there's 0a218e04fe70be7696001f6db7f6783b5d829944 as a nice though short example. Another good example is `src/session/__tests__/sessionReducer-test.js`. And a longer example, covering more complexity than we need in this file, is `src/unread/__tests__/unreadModel-test.js`.

Reply:
Done.